[Index Management] Few UX improvements

sabarasaba opened the pull request:
Summary

This PR addresses two small issues of the creation flow:

* Since the data for each step is only saved into the draft policy upon clicking the `next` button, we should disable clicking the wizard steps. This also means that we are introducing a back button for step 2 and 3.
* If `create and execute policy` fails executing, the request fails and user is still in the creation flow but they cannot create the policy because it was already created! So now we check if executing fails, and if so we delete the previously created policy as a teardown process.


<details>
<summary>Screenshots</summary>

// Back buttons
![Screenshot 2023-09-17 at 10 45 09](https://github.com/elastic/kibana/assets/1191206/480e1725-aad8-4229-957e-a1e7adc78d68)
![Screenshot 2023-09-17 at 10 45 16](https://github.com/elastic/kibana/assets/1191206/a31e751c-8559-461a-b242-436e845d195a)

// Breadcrumbs
![Screenshot 2023-09-17 at 11 07 09](https://github.com/elastic/kibana/assets/1191206/03232f1f-15bd-45f4-af60-27592480b009)
![Screenshot 2023-09-17 at 11 07 13](https://github.com/elastic/kibana/assets/1191206/eb3e4999-79ca-4ff3-9401-5b0de3080b4a)


</details>

elasticmachine commented:
Pinging @elastic/platform-deployment-management (Team:Deployment Management)

alisonelizabeth commented on x-pack/plugins/index_management/public/application/sections/enrich_policy_create/create_policy_wizard.tsx:
```diff
@@ -93,7 +94,7 @@ export const CreatePolicyWizard = () => {
         }),
         status: completionState.configurationStep ? 'complete' : getStepStatus(CONFIGURATION),
-        onClick: () => currentStep !== CONFIGURATION && changeCurrentStepTo(CONFIGURATION),
+        onClick: noop,
```
Is this necessary? Is `onClick` required?

alisonelizabeth reviewed (commented):
Thanks for working on this @sabarasaba!

For some reason, I thought there was a "Previous" or "Back" link to each step. Now that we've disabled the step-based navigation, there is not an easy way to go back a step. For example, in this state, if I wanted to fix a field, I would need to start over again. Sorry I didn't realize this when we had the initial discussion. Let me know what you think.

<img width="618" alt="Screenshot 2023-09-13 at 4 08 46 PM" src="https://github.com/elastic/kibana/assets/5226211/4378adc7-63d1-45c6-be20-59afb1e48fef">

alisonelizabeth commented:
Also, not directly related to this PR, but something I noticed while testing - we'll want to make similar fixes to the breadcrumbs that Yulia did in https://github.com/elastic/kibana/pull/165894.

sabarasaba commented:
@elasticmachine merge upstream

sabarasaba commented:
Thanks for checking it out @alisonelizabeth! I've added support for breadcrumbs, similar to the work that yulia started, and also added back button support.

alisonelizabeth commented on x-pack/plugins/index_management/public/application/services/breadcrumbs.ts:
```diff
@@ -70,11 +78,21 @@ class BreadcrumbService {
+      ...this.breadcrumbs.enrichPolicies,
+      {
+        text: i18n.translate('xpack.idxMgmt.breadcrumb.enrichPolicyCreateLabel', {
+          defaultMessage: 'Create enrich policies',
```
[nit] Should the breadcrumb copy align with the page title?

```suggestion
          defaultMessage: 'Create enrich policy',
```

<img width="651" alt="Screenshot 2023-09-18 at 12 51 38 PM" src="https://github.com/elastic/kibana/assets/5226211/9f48a27c-5324-4943-bb93-691981f7311c">

alisonelizabeth reviewed (approved):
Latest LGTM. Left one small copy suggestion for the breadcrumb.
